### PR TITLE
Adding examples for Type Inference

### PR DESCRIPTION
**Pull request by HillmerCh**

### CODE REVIEW

File: session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+**Local Variable Type Inference for Primitive Data Types example**
+
+This first example is a class named `PrimitiveTypes` wichs its methods should use
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md]: wichs its methods ==> whose methods

File: session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+**Local Variable Type Inference for Primitive Data Types example**
+
+This first example is a class named `PrimitiveTypes` wichs its methods should use 
+`Type Inference` to declare variables using them in Expressions, Statements, and Blocks as Loops.
+
+Perform the below commands to see the contents of the respective Java classes contained in the src folder:
+
+    $ cat src/com/typeinference/PrimitiveTypes.java
+
+**Note:** in case one of the below .sh script fails due to the tree command, please take a look at Download and install the tree and wget command section in the README.md file and apply the appropriate solution.
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md]: try to put keywords like `tree` etc in backticks (`) also `.sh` and README.md

File: session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+Compiled the example with the following command:
+
+    $ ./compile.sh
+
+And we run the example with the following command:
+
+    $ ./run.sh
+    
+Now, open the file `com/typeinference/PrimitiveTypes.java` and replace all primitive data types declaration with
+the new keyword `var`, this change allow you to use Variable Type Inference.
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md]: allow ==> allows

File: session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    int id --> var id
+
+Once you have replaces all primitive data types declaration, execute the commands for compile and run.
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md]: replaces ==> replaced

File: session-1-Local-Variable-Type-Inference/02_Object_Data_Types/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+**Local Variable Type Inference for Object Data Types example**
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/02_Object_Data_Types/README.md:1]: Apply same corrections from the above to this section as well

File: session-1-Local-Variable-Type-Inference/03_Arrays_Data_Types/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+**Local Variable Type Inference for Arrays Data Types example**
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/03_Arrays_Data_Types/README.md:1]: Same corrections

File: session-1-Local-Variable-Type-Inference/04_Full_uses/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+**Local Variable Type Inference for all Types example**
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/04_Full_uses/README.md:1]: Ditto

File: session-1-Local-Variable-Type-Inference/05_JShell/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+**Local Variable Type Inference for all Types example**
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/05_JShell/README.md:1]: Ditto

File: session-1-Local-Variable-Type-Inference/05_JShell/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+     jshell
+     
+Verify you are using JShell for Java 10 version
+
+    |  Welcome to JShell -- Version 10
+    |  For an introduction type: /help intro
+
+
+To execute JShell online go to next URL
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/05_JShell/README.md:17]: Add link to a whole array of java shells as shared at https://github.com/neomatrix369/jdk9-jigsaw/tree/master/session-3-jshell (see whole page for links)

**Comment by neomatrix369** [comment]
Very good start and looks good, do you also have the examples branch ready?

Also, I'd like to see some edge cases with the usage of `var` type.

For e.g.

```java
		var aVarVariable = "String";
		System.out.println(aVarVariable);

		aVarVariable = "Changed value";   // <=== allowed, compiles, similar to Scala's `var`?
		System.out.println(aVarVariable);
```

or

```java
		aVarVariable = 10;   // <=== not allowed, compiler error
		System.out.println(aVarVariable);
```
>> Java10var.java:9: error: incompatible types: int cannot be converted to String
>>		aVarVariable = 10;   // <=== not allowed, compiler error
>>		               ^

or

```java
		var aVarVariable = "Change value again";   // <=== compiler error
		System.out.println(aVarVariable);
```
>> Java10var.java:12: error: variable aVarVariable is already defined in method main(String[])
>> 		var aVarVariable = "Change value again";   // <=== compiler error
>>		    ^

or

```java
		var aVarVariable = 10;   // <=== compiler error
		System.out.println(aVarVariable);
```
>> Java10var.java:15: error: variable aVarVariable is already defined in method main(String[])
>> 		var aVarVariable = 10;   // <=== compiler error
>>		    ^


The above is good for both as an exercise (guess what happens when you compile it) and also as examples as to this is what happens when you compile this block of code in Java 10 using `var`.

**Review by neomatrix369 — CHANGES_REQUESTED** [review]
Overall looks good, some suggestions were made and also some additional aspects

**Comment by neomatrix369** [comment]
```java
public class Java10var {
	public static void main(String args[]) {
		var aVarVariable = "String";
		System.out.println(aVarVariable);

		aVarVariable = "Changed value";   // <=== allowed, compiles
		System.out.println(aVarVariable);

		aVarVariable = 10;   // <=== not allowed, compiler error
		System.out.println(aVarVariable);

		var aVarVariable = "Change value again";   // <=== compiler error
		System.out.println(aVarVariable);

		var aVarVariable = 10;   // <=== compiler error
		System.out.println(aVarVariable);
	}
}
```

Here's my block of code I used as an example.

**Comment by neomatrix369** [comment]
@HillmerCh :+1: good work!

File: session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md (hunk 1 above)
Comment (HillmerCh) [reply session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md]: Done

File: session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md (hunk 2 above)
Comment (HillmerCh) [reply session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md]: Done

File: session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md (hunk 3 above)
Comment (HillmerCh) [reply session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md]: Done

File: session-1-Local-Variable-Type-Inference/02_Object_Data_Types/README.md (hunk 5 above)
Comment (HillmerCh) [reply session-1-Local-Variable-Type-Inference/02_Object_Data_Types/README.md:1]: Done

File: session-1-Local-Variable-Type-Inference/03_Arrays_Data_Types/README.md (hunk 6 above)
Comment (HillmerCh) [reply session-1-Local-Variable-Type-Inference/03_Arrays_Data_Types/README.md:1]: Done

File: session-1-Local-Variable-Type-Inference/04_Full_uses/README.md (hunk 7 above)
Comment (HillmerCh) [reply session-1-Local-Variable-Type-Inference/04_Full_uses/README.md:1]: Done

File: session-1-Local-Variable-Type-Inference/05_JShell/README.md (hunk 8 above)
Comment (HillmerCh) [reply session-1-Local-Variable-Type-Inference/05_JShell/README.md:1]: Done

File: session-1-Local-Variable-Type-Inference/05_JShell/README.md (hunk 9 above)
Comment (HillmerCh) [reply session-1-Local-Variable-Type-Inference/05_JShell/README.md:17]: Done

**Comment by HillmerCh** [comment]
@neomatrix369, thanks for your corrections. I have the examples but in only one class, this morning I decided to split the examples in 5,  I will be working on split my class. I will pull them this weekend.
The example 04  is  the same as you say in your class `Java10var`, the main idea is get compile errors so everyone must be go back to the class and fix them.
In the example 05 (JShell) I think to add some errors too

File: session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
+**Local Variable Type Inference for Primitive Data Types example**
+
+This first example is a class named `PrimitiveTypes` whose methods should use 
+`Type Inference` to declare variables using them in Expressions, Statements, and Blocks as Loops.
+
+Perform the below commands to see the contents of the respective Java classes contained in the src folder:
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md:6]: make things like src ==> `src`

File: session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+    $ cat src/com/typeinference/PrimitiveTypes.java
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/01_Primitive_Data_Types/README.md:8]: within ``` so it looks highlighted

File: session-1-Local-Variable-Type-Inference/04_Full_uses/README.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+This example is a class named `FullUsesTypes` wichs its methods should use
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/04_Full_uses/README.md:3]: which is misspelt, same issue everywhere

File: session-1-Local-Variable-Type-Inference/05_JShell/README.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    |  For an introduction type: /help intro
+    
+    
+Once you are on JShell, try to do next excercises using Variable Type Inference
+
+- Make a sum of two number of the type int, and print the result
+- Declarate text using String, StringBuilder, StringBuffer
+- Declarate collections as List, ArrayList, Set, Map
+
+For Object type you can use the method getClass() to check the type that the compiler used.
```
Comment (neomatrix369) [inline session-1-Local-Variable-Type-Inference/05_JShell/README.md:33]: getClass() and tokens like this should be in ``` backticks

**Comment by neomatrix369** [comment]
Merging now but please consider the corrections mentioned and raise a new PR @HillmerCh

**Pull request merged**